temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivitySlotSupplierQueue.java

```diff
@@ -73,13 +72,13 @@ static final class QueuedLARequest {
               return 0;
             });
     this.slotSupplier = slotSupplier;
-    this.queueThread = new Thread(this::processQueue, "LocalActivitySlotSupplierQueue");
-    this.queueThread.start();
+    this.queueThreadService =
```

Comment:
Note I believe this will always create a thread even if users are not using local activities. To be more proper we would only create the executor if the worker is started. Originally I said that wasn't needed with the thread impl., but with a Service it is different.

Reply:
It does not create a thread until you submit the first task: https://stackoverflow.com/questions/43569407/when-does-executor-really-create-new-thread

Reply:
Hmm OK that goes against my reading of the java docs, but source code is king